src/loaders/s3m_load.c

```diff
@@ -284,14 +284,15 @@ static int s3m_load(struct module_data *m, HIO_HANDLE * f, const int start)
 	mod->chn = 0;
 
 	for (i = 0; i < 32; i++) {
+		int x;
 		if (sfh.chset[i] == S3M_CH_OFF)
 			continue;
 
 		mod->chn = i + 1;
 
-		if (sfh.mv & 0x80) {	/* stereo */
-			int x = sfh.chset[i] & S3M_CH_PAN;
-			mod->xxc[i].pan = (x & 0x0f) < 8 ? 0x30 : 0xc0;
+		x = sfh.chset[i] & S3M_CH_NUMBER;
+		if ((sfh.mv & 0x80) && x < S3M_CH_ADLIB) {	/* stereo */
+			mod->xxc[i].pan = x < S3M_CH_RIGHT ? 0x30 : 0xc0;
```

Comment:
Probably the part I'm least sure about—it seems correct for ST3 and OpenMPT, but MPT 1.16 will foolishly emit AdLib channels for channels past 16. It also emits default panning, so I don't think this affects much...

Reply:
Yes, MPT always writes a pan table, so in practice this doesn't matter.